### PR TITLE
refactor(xrpl): transactions

### PR DESCRIPTION
**Pull request by GuillemGarciaDev**
# refactor(xrpl): transactions

This PR aims to upgrade the transaction models for a better usage. BaseTx is still used with a new method `Flatten` to convert transactions from go structs to desired pre-encoding format.

## Changes

- Moved `AMMXX` transactions from `tx.go` to its own file with corresponding test.
- Implemented `Flatten` method on `Payment`, `Clawback`, `TicketCreate` and `TrustSet`
- Declared `Flatten` method on every transaction model
- Updated `wallet` examples
- Added `transactions/payment` example

## Notes (optional)

Before releasing the first version of the sdk, it is necessary to implement and test every `Flatten` method on transaction models.

On a shorter term, `xrpl/models/transactions/tx.go` should be refactored to split responsabilities on interface declaration and marshalling.

### CODE REVIEW

File: xrpl/model/transactions/payment.go [hunk 1]
```diff
@@ -57,8 +57,46 @@ type Payment struct {
 	SendMax types.CurrencyAmount `json:",omitempty"`
 }
 
-func (*Payment) TxType() TxType {
-	return PaymentTx
+func (p *Payment) Flatten() map[string]interface{} {
```
Comment (florent-uzio) [inline xrpl/model/transactions/payment.go:60]: So am I correct to say that `Flatten` converts a `struct` to a `map[string]interface{}` ?

File: xrpl/model/transactions/trust_set_test.go [hunk 2]
```diff
@@ -51,3 +51,42 @@ func TestTrustSetTx(t *testing.T) {
...
+			Value:    "100",
+		},
+	}
+
+	flattened := s.Flatten()
+	
+	expected := map[string]interface{}{
+		"Account":            "ra5nK24KXen9AHvsdFTKHSANinZseWnPcX",
+		"TransactionType":    "TrustSet",
+		"Fee":                uint64(12),
```
Comment (florent-uzio) [inline xrpl/model/transactions/trust_set_test.go:77]: Shouldn't Fee be a string according to the docs ?
https://xrpl.org/docs/references/protocol/transactions/common-fields#transaction-common-fields

**Review by florent-uzio — APPROVED** [review]
Looks great! 
I have a question regarding the Fee.

File: xrpl/model/transactions/trust_set_test.go (hunk 2 above)
Comment (GuillemGarciaDev) [reply xrpl/model/transactions/trust_set_test.go:77]: Nice catch!

I checked different transaction models types and there are some mismatches with the transaction docs you mentioned. 

I have updated the desired ones (`Payment`, `Clawback`, `TrustSet` and `TicketCreate`)

File: xrpl/model/transactions/tx.go [hunk 3]
```diff
@@ -26,119 +27,127 @@ func (tx *Binary) TxType() TxType {
...
-	BaseTx
-}
+	if tx.Account != "" {
+		flattened["Account"] = tx.Account.String()
+	}
+	if tx.TransactionType != "" {
+		flattened["TransactionType"] = tx.TransactionType
+	}
+	if tx.Fee != 0 {
+		flattened["Fee"] = tx.Fee.String()
```
Comment (florent-uzio) [inline xrpl/model/transactions/tx.go:114]: Nice update

**Review by florent-uzio — APPROVED** [review]
Looks great, thanks for updating the Fee type.

**Pull request merged**